scripts/core/exp-command-buffer.yml

```diff
@@ -1203,11 +1204,15 @@ params:
       desc: "[in] Struct defining how the kernel command is to be updated."
 returns:
     - $X_RESULT_ERROR_UNSUPPORTED_FEATURE:
-        - "If update functionality is not supported by the device."
+        - "If $X_DEVICE_COMMAND_BUFFER_UPDATE_CAPABILITY_FLAG_KERNEL_ARGUMENTS is not supported by the device, but any of `pUpdateKernelLaunch->numNewMemObjArgs`, `pUpdateKernelLaunch->numNewPointerArgs`, or `pUpdateKernelLaunch->numNewValueArgs` are not zero."
+        - "If $X_DEVICE_COMMAND_BUFFER_UPDATE_CAPABILITY_FLAG_LOCAL_WORK_SIZE is not supported by the device but `pUpdateKernelLaunch->pNewLocalWorkSize` is not nullptr."
```

Comment:
In the update descriptor wording we have the following:

>  If `pNewGlobalWorkSize` is set and `pNewLocalWorkSize` is nullptr,
        then the runtime implementation will choose the local work size.

If the device doesn't support `UR_DEVICE_COMMAND_BUFFER_UPDATE_CAPABILITY_FLAG_LOCAL_WORK_SIZE` are we just assuming that the implementation won't generate a new local work size here (which may be different from the previous value) or that it would do that but throw an error because updating the local size is not possible?

If the former we should probably change that quoted wording to reflect that it is not guaranteed to generate a new local size if not supported, or document the latter error here.

Reply:
Good shout, I went with an error because the old local size might not be compatable with the new global size if we can't update it under the hood.